src/components/LeaderBoard/Leaderboard.jsx

```diff
@@ -184,8 +184,8 @@ const LeaderBoard = ({
             </tr>
             {leaderBoardData.map((item, key) => (
               <tr key={key}>
-                <td className="align-middle">
-                  <Link to={`/dashboard/${item.personId}`}>
+                <td className="align-middle" onClick={() => { if (window.confirm(`Are you sure you wish to view this ${item.name} dashboard ?`)) {window.open(`/dashboard/${item.personId}`, "Popup","toolbar=no, location=no, statusbar=no, menubar=no, scrollbars=1, resizable=0, width=580, height=600, top=30")}} }> 
```

Comment:
You can write a function to be called onClick event for the table row, and insert the conditional logic in the function body. Also, please make use of [<Redirect>](https://v5.reactrouter.com/web/api/Redirect) from react-router to navigate to a different view. 'history' object is also an alternative, you can use history.push() to route the location of the user, but IMO it is preferable to route using react-router which is being used in the app.